packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -39,15 +44,44 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
     style = variant === 'accent' || variant === 'cta' ? 'fill' : 'outline',
     staticColor,
     isDisabled,
+    isPending,
     autoFocus,
     ...otherProps
   } = props;
   let domRef = useFocusableRef(ref);
   let {buttonProps, isPressed} = useButton(props, domRef);
```

Comment:
According to design, I think the button should stop responding to clicks during the delay before the spinner appears, even if it still looks enabled. So you might need to do something like this:

```js
useButton({
  ...props,
  isDisabled: props.isDisabled || props.isPending
}, domRef)
```

https://spectrum.adobe.com/page/button/#Delay-before-pending-state

Reply:
I ended up writing a small hook to filter out the press and click events while pending since the disabled visual state is closely tied to the disabled attribute. I added aria-disabled as well.

Reply:
Wouldn't this mean the pressed state would still apply even if the events are not triggered? Did you try getting rid of the `:disabled` selector and just using `is-disabled` for the visual state?